### ISSUE TITLE
componentWillUnmount would net been called in React-native Navigation

### ISSUE THREAD

**Issue by jiataocai** [dsuryd/dotNetify#188]
Hello, Dicky , DotNetify is a great solution and I have used it in my React-Native application both Android and IOS. but I found componentWillUnmount have never been called when use StackNavigator, as a result, "this.vm.$destroy()" also have never been called , I have to destroy the vm when the user leave the current screen. I know maybe I can use Navigation lifecycle listener to resolve the issue:

componentDidMount() {
  this.didFocusSubscription = this.props.navigation.addListener(
    'willFocus', () => { this.vm.$destroy()}
  );
}
componentWillUnmout() {
    this.didFocusSubscription.remove();
}
  }

but not sure if this is the best approach to use DotNetify in native.
could you give me some suggestion or best practice?

Thanks,

**Comment by dsuryd** [comment]
Hi, how do you reconnect if the user navigates back?

I did a similar approach in this demo: https://github.com/dsuryd/dotnetify-react-native-demo/blob/master/src/screens/LiveGaugeScreen.js

Basically, there's a global object that tracks active screen and emits event when it changes.  The view subscribes to this and makes the decision to connect or destroy based on the info.

**Comment by jiataocai** [comment]
Hi, thank you so much for your quick response.
I have tested the demo you mentioned before, and found that it will show an error message if I click "Login" and navigate to LiveGaugeScreen directly but not AFI100Screen :
"Component is attempting to connect to an already active 'LiveGaugeVM'. if it's from a dismounted component..."
you can reproduce the error by just changing:
const tabNavigation = TabNavigator(
  {
    AFITop100: { screen: AFITop100Screen },
    LiveGauge: { screen: LiveGaugeScreen }
  },
to:
const tabNavigation = TabNavigator(
  {    
        LiveGauge: { screen: LiveGaugeScreen },
        AFITop100: { screen: AFITop100Screen }
  },

in AppNavigation.js.

I did some investigation and found that the ScreenTracker.setScreen('LiveGauge') and   handleNavigationStateChange(prevState, newState)  will be called after login, so it will navigate twice LiveGaugeScreen and cause the error.

**Comment by dsuryd** [comment]
Would you like to fix the logic and send me a PR?  :)

**Comment by jiataocai** [comment]
I love to do:)
I added a conditional check before dotnetify.react.connect in the LiveGaugeScreen:

connectLiveGauge = screen => {
    const self = this;
      if (screen == 'LiveGauge') {
          **_if (dotnetify.react.viewModels.hasOwnProperty('LiveGaugeVM')) {
              this.vm.$destroy();
              this.vm = null;
          };_**
      this.vm = dotnetify.react.connect('LiveGaugeVM', this, {
        exceptionHandler: ex => ScreenTracker.goToLoginScreen(self.navigate, ex)
      });
    }
    else if (this.vm) {
      this.vm.$destroy();
      this.vm = null;
    }
  };

and it can resolve the issue, but not sure if it is the best solution.

**Comment by dsuryd** [comment]
I don't think you need to destroy the `vm` ; if it's already initialized, just keep it:
```jsx
if (screen == 'LiveGauge') {
   this.vm = this.vm || dotnetify.react.connect('LiveGaugeVM', ...
```

Although it's best to make `ScreenTracker` not firing on the same screen twice, and avoid adding that extra condition in every screen.
